### PR TITLE
Convert to FastAPI

### PR DESCRIPTION
**Pull request by MoritzHahn1337**
```
Added FastAPI Adapter (GDEV-1317)
Used httpyexcept exception handling (GDEV-1318)
Custom Title in Swagger UI (GDEV-1444)

Co-authored-by: Thomas Zajac <thomas-jakob.zajac@uni-tuebingen.de>
Co-authored-by: KerstenBreuer <kersten-breuer@outlook.com>
```

### CODE REVIEW

File: dcs/adapters/outbound/event_broadcast.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+class DrsEventBroadcaster(DrsEventBroadcasterPort):
+    """An implementation for the DrsEventBroadcasterPort."""
+
+    async def download_served(self, *, drs_object: models.DrsObjectWithUri) -> None:
+        """Communicate the event of an download being served. This can be relevant for
+        auditing purposes."""
+
+        pass  # pylint: disable=unnecessary-pass
```
Comment (mephenor) [inline dcs/adapters/outbound/event_broadcast.py:29]:
```suggestion
```

File: dcs/adapters/outbound/event_broadcast.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    async def unstaged_download_requested(
+        self, *, drs_object: models.DrsObjectWithUri
+    ) -> None:
+        """Communicates the event that a download was requested for a DRS object, that
+        is not yet available in the outbox."""
+
+        pass  # pylint: disable=unnecessary-pass
```
Comment (mephenor) [inline dcs/adapters/outbound/event_broadcast.py:37]:
```suggestion
```

File: dcs/adapters/outbound/event_broadcast.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    async def new_drs_object_registered(
+        self, *, drs_object: models.DrsObjectWithUri
+    ) -> None:
+        """Communicates the event that a file has been registered."""
+
+        pass  # pylint: disable=unnecessary-pass
```
Comment (mephenor) [inline dcs/adapters/outbound/event_broadcast.py:44]:
```suggestion
```

File: dcs/adapters/outbound/event_broadcast.py (hunk 1 above)
Comment (mephenor) [reply dcs/adapters/outbound/event_broadcast.py:29]: you don't nedd `...` or  `pass`  if you have docstrings.

File: dcs/main.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    container.config.load_config(config)
+
+    return container
+
+
+def get_rest_api(*, config: Config) -> FastAPI:
+    """Creates a FastAPI app.
+    For full functionality of the api, run in the context of an CI container with
+    correct wireing and initialized resources (see the run_api function below).
+    """
```
Comment (mephenor) [inline dcs/main.py:39]:
```suggestion
    """
    Creates a FastAPI app.
    For full functionality of the api, run in the context of an CI container with
    correct wireing and initialized resources (see the run_api function below).
    """
```

File: tests/fixtures/config.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+) -> Config:
+    """Merges parameters from the default TEST_CONFIG_YAML with params inferred
+    from testcontainers."""
+    sources_dict: Dict[str, object] = {}
+
+    if sources is not None:
+        for source in sources:
+            sources_dict.update(**source.dict())
+
+    return Config(config_yaml=default_config_yaml, **sources_dict)  # type: ignore
```
Comment (mephenor) [inline tests/fixtures/config.py:41]:
```suggestion
    merged_sources: Dict[str, object] = {}

    if sources is not None:
        for source in sources:
            merged_sources.update(**source.dict())

    return Config(config_yaml=default_config_yaml, **merged_sources)  # type: ignore
```

File: tests/test_typical_journey.py [hunk 6]
```diff
@@ -78,28 +78,26 @@ async def test_happy(
...
-    drs_object_with_access = await data_repo.access_drs_object(drs_id=drs_object.id)
+    drs_object_response = await joint_fixture.rest_client.get(
+        f"/objects/{drs_object.id}"
+    )
 
     # download file bytes:
-    response = requests.get(drs_object_with_access.access_url)
-    response.raise_for_status()
-    assert response.content == file_object.content
+    request_response = requests.get(drs_object_response.json()["access_url"])
```
Comment (mephenor) [inline tests/test_typical_journey.py:101]:
```suggestion
    access_url = drs_object_response.json()["access_url"]
    request_response = requests.get(access_url)
```

**Review by mephenor — APPROVED** [review]

**Review by mephenor — APPROVED** [review]

**Review by KerstenBreuer — DISMISSED** [review]
Sieht gut aus

**Review by mephenor — APPROVED** [review]

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**